apps/aecore/test/aec_conductor_tests.erl

```diff
@@ -373,17 +379,147 @@ test_block_publishing() ->
+             TmpKeysDir
+     end,
+     fun(TmpKeysDir) ->
+             teardown_common(TmpKeysDir),
+             meck:unload(aec_headers),
+             meck:unload(aec_blocks),
+             ok
+     end,
+     [
+        {"Start signing after mined block", fun test_mined_block_signing/0},
+        {"Start signing after two mined block", fun test_two_mined_block_signing/0},
+        {"Start signing after received block", fun test_received_block_signing/0}
+     ]}.
+
+test_mined_block_signing() ->
+    Keys = beneficiary_keys(),
+    true = aec_events:subscribe(block_created),
+    true = aec_events:subscribe(micro_block_created),
+
+    ?TEST_MODULE:start_mining(),
+    assert_leader(false),
+    assert_generation_state(stopped),
+    KeyBlock = wait_for_block_created(),
+    assert_leader(true),
+    assert_generation_state(running),
+
+    ok = aec_tx_pool:push(tx(Keys)),
+
+    MicroBlock = wait_for_micro_block_created(),
+    {ok, KeyHash} = aec_blocks:hash_internal_representation(KeyBlock),
+    ?assertEqual(KeyHash, aec_blocks:prev_key_hash(MicroBlock)),
+    ok.
+
+test_two_mined_block_signing() ->
+    Keys = beneficiary_keys(),
+    true = aec_events:subscribe(block_created),
+    true = aec_events:subscribe(micro_block_created),
+
+    ?TEST_MODULE:start_mining(),
+    assert_leader(false),
+    assert_generation_state(stopped),
+    KeyBlock1 = wait_for_block_created(),
+    assert_leader(true),
+    assert_generation_state(running),
+    KeyBlock2 = wait_for_block_created(),
+    assert_leader(true),
+    assert_generation_state(running),
+
+    ok = aec_tx_pool:push(tx(Keys)),
+
+    MicroBlock = wait_for_micro_block_created(),
+    {ok, KeyHash1} = aec_blocks:hash_internal_representation(KeyBlock1),
+    {ok, KeyHash2} = aec_blocks:hash_internal_representation(KeyBlock2),
+    ?assertEqual(KeyHash1, aec_blocks:prev_key_hash(KeyBlock2)),
+    ?assertEqual(KeyHash2, aec_blocks:prev_key_hash(MicroBlock)),
+    ok.
+
+test_received_block_signing() ->
+    Keys = beneficiary_keys(),
+    meck:new(aec_mining, [passthrough]),
+    meck:expect(aec_mining, mine,
+                fun(_, _, _) -> timer:sleep(1000), {error, no_solution} end),
+    true = aec_events:subscribe(block_to_publish),
+
+    ?TEST_MODULE:start_mining(),
+    assert_leader(false),
+    assert_generation_state(stopped),
+
+    [_GB, KB1] = aec_test_utils:gen_blocks_only_chain(2, preset_accounts(Keys)),
+    ?assertEqual(ok, ?TEST_MODULE:post_block(KB1)),
+
+    KB1 = wait_for_block_to_publish(),
+    ?assertEqual(true, aec_blocks:is_key_block(KB1)),
+    assert_leader(true),
+    assert_generation_state(running),
+
+    %% Single tx should trigger micro block
+    ok = aec_tx_pool:push(tx(Keys)),
+
+    meck:unload(aec_mining),
+
+    NewBlock = wait_for_block_to_publish(),
+    ?assertEqual(false, aec_blocks:is_key_block(NewBlock)),
+    {ok, KB1Hash} = aec_blocks:hash_internal_representation(KB1),
+    ?assertEqual(KB1Hash, aec_blocks:prev_key_hash(NewBlock)),
```

Comment:
It seems to me that the test shall be integrated with assertions that:
* The signed microblock `NewBlock` is inserted in the chain.
* The preceding posted keyblock `KB1` is inserted in the chain.

It may be constrained by how the code is written, though from a test perspective it is an independent check that would ease readability of test (unless test is explicitly not meant to store blocks in chain for some reason?). The benefit that I have in mind is the following checks having passed on these blocks:
* [Microblock signature verifiable according to pubkey in prev key block](https://github.com/aeternity/epoch/blob/8d69cfd2/apps/aecore/src/aec_chain_state.erl#L429).
* [Connection to genesis](https://github.com/aeternity/epoch/blob/8d69cfd2/apps/aecore/src/aec_chain_state.erl#L419).